geospace-lib/src/main/scala/com/socrata/geospace/lib/shapefile/LayerTransformer.scala

```diff
@@ -0,0 +1,117 @@
+package com.socrata.geospace.lib.shapefile
+
+import java.io.File
+
+import com.socrata.geospace.lib.Utils._
+import com.socrata.geospace.lib.errors.InvalidShapefileSet
+import com.typesafe.scalalogging.slf4j.Logging
+import org.apache.commons.io.FilenameUtils
+import org.geoscript.layer.Shapefile
+import org.geoscript.projection.{Projection, _}
+import org.geoscript.feature._
+import org.geoscript.layer._
+import org.geoscript.projection._
+import org.geotools.referencing.ReferencingFactoryFinder
+import org.opengis.referencing.crs.CoordinateReferenceSystem
+import org.geoscript.feature.schemaBuilder._
+import org.geotools.factory.Hints
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Shape file reader.
+ *
+ * <p> Note: As currently used each layer (distinguished by their namespace) will
```

Comment:
Actually, the intent of this class is to take one layer, make transformations, and return values. It does not retain any state/values of any particular layer. It must be initialized with constant necessary to make parsing/transforming possible (in particular the projection details).